packages/amazon-bedrock/src/bedrock-chat-language-model.test.ts

```diff
@@ -994,22 +1112,76 @@ describe('doGenerate', () => {
     };
     stopReason?: string;
     trace?: typeof mockTrace;
+    reasoningContent?:
+      | {
+          reasoningContent: {
+            reasoningText: {
+              text: string;
+              signature?: string;
+            };
+          };
+        }
+      | {
+          reasoningContent: {
+            redactedReasoning: {
+              data: string;
+            };
+          };
+        }
+      | Array<
+          | {
+              reasoningContent: {
+                reasoningText: {
+                  text: string;
+                  signature?: string;
+                };
+              };
+            }
+          | {
+              reasoningContent: {
+                redactedReasoning: {
+                  data: string;
+                };
+              };
+            }
+        >;
   }) {
+    // Map the content array to the format expected by the API
+    const contentBlocks = content.map(item => {
+      if ('type' in item) {
+        if (item.type === 'text') {
+          return { text: item.text };
+        } else if (item.type === 'thinking') {
+          return {
+            reasoningContent: {
+              reasoningText: {
+                text: item.thinking,
+                signature: item.signature,
+              },
+            },
+          };
+        } else if (item.type === 'tool_use') {
+          return {
+            toolUse: {
+              toolUseId: item.id,
+              name: item.name,
+              input: item.input,
+            },
+          };
+        }
+      } else if ('reasoningContent' in item) {
+        return item;
+      }
+      return item;
+    });
```

Comment:
this looks a bit unclear (mixing `in` w/ type). why is this necessary? can this be solved in a switch? if not, consider adding some comments

Reply:
[d85704e](https://github.com/vercel/ai/pull/4980/commits/d85704e65a68d8341f67440559037fc46a1c4d2f)

I was under the assumption that the content needed remapping, that's not the case